drivers/Timer.h

```diff
@@ -92,21 +97,27 @@ class Timer : private NonCopyable<Timer> {
 
     /** An operator shorthand for read()
      */
+    MBED_DEPRECATED_SINCE("mbed-os-6.0.0", "Floating point operators should normally be avoided for code size. If really needed, you can use `duration<float>(read_duration()).count()`")
     operator float();
 
     /** Get in a high resolution type the time passed in microseconds.
      *  Returns a 64 bit integer.
      */
     us_timestamp_t read_high_resolution_us();
 
+    /** Get in a high resolution type the time passed in microseconds.
```

Comment:
What's the design choice behind leaving `read_us` and `read_ms`?

Reply:
It's me being soft. I am inclined to remove them, but that forces people to understand how to do `duration_cast<milliseconds>(t.read_duration()).count()` for what is probably a fairly common operation (at least in the sort of glue code where this is used to provide a "HAL" for something using plain ints).

I only went half-way and removed the `float` form, as I didn't want floats to be used casually.

If you're okay with removing `read_us` and `read_ms`, I am too.

Reply:
I think it's fair to get rid of it now. I don't really like the `read_duration` I'd really like to have it just `read` especially that we are deprecating the float read(). I know it's against our own rules but for the sake of clarity and cleanliness maybe it's worth just replacing them?

Reply:
I'm also prefer to have something short like `read`.  `read` is generic, so I'm not a huge fan. Read what? Is there something better and short? It's not `now`, because it's not a `time_point`. 

Maybe `elapsed_time`? That is what `time_point`s have to return their `duration`. This isn't a time point, with an elapsed real-time since an epoch, but it *does* measure elapsed time like a stop-watch (with start/stop and reset functions). Still a bit long though.